test/credentials.spec.coffee

```diff
@@ -254,17 +262,127 @@ if AWS.util.isNode()
         validateCredentials(creds, 'RELOADED', 'RELOADED', 'RELOADED')
 
       it 'fails if credentials are not in the file', ->
         mock = ''
         helpers.spyOn(AWS.util, 'readFileSync').andReturn(mock)
 
         new AWS.SharedIniFileCredentials().refresh (err) ->
-          expect(err.message).to.equal('Credentials not set in /home/user/.aws/credentials using profile default')
+          expect(err.message).to.equal('Profile default not found in /home/user/.aws/credentials')
 
         expect(-> new AWS.SharedIniFileCredentials().refresh()).
-          to.throw('Credentials not set in /home/user/.aws/credentials using profile default')
+          to.throw('Profile default not found in /home/user/.aws/credentials')
+
+  describe 'loadRoleProfile', ->
+    beforeEach -> 
+      process.env.HOME = '/home/user'
+    
+    it 'will fail if assume role is disabled', ->
+      mock = '''
+      [default]
+      aws_access_key_id = akid
+      aws_secret_access_key = secret
+      role_arn = arn
+      '''
+      helpers.spyOn(AWS.util, 'readFileSync').andReturn(mock)
+      creds = new AWS.SharedIniFileCredentials({disableAssumeRole: true})
+      creds.refresh (err) ->
+        expect(err.message).to.equal('Role assumption profiles are disabled. Failed to load profile default from /home/user/.aws/credentials')
+      
+    it 'will fail if no source profile is specified', ->
+      mock = '''
+      [default]
+      aws_access_key_id = akid
+      aws_secret_access_key = secret
+      role_arn = arn
+      '''
+      helpers.spyOn(AWS.util, 'readFileSync').andReturn(mock)
+    
+      creds = new AWS.SharedIniFileCredentials()
+      creds.refresh (err) ->
+        expect(err.message).to.equal('source_profile is not set in /home/user/.aws/credentials using profile default')
+    
+    it 'will fail if source profile config is not defined', ->
+      mock = '''
+      [default]
+      aws_access_key_id = akid
+      aws_secret_access_key = secret
+      role_arn = arn
+      source_profile = fake
+      '''
+      helpers.spyOn(AWS.util, 'readFileSync').andReturn(mock)
+      
+      creds = new AWS.SharedIniFileCredentials()
+      creds.refresh (err) ->
+        expect(err.message).to.equal('source_profile fake set in /home/user/.aws/credentials using profile default does not exist')
+    
+    it 'will fail if source profile config lacks credentials', ->
+      mock = '''
+      [default]
+      aws_access_key_id = akid
+      aws_secret_access_key = secret
+      role_arn = arn
+      source_profile = foo
+      [foo]
+      aws_access_key_id = akid2
+      '''
+      helpers.spyOn(AWS.util, 'readFileSync').andReturn(mock)
+      
+      creds = new AWS.SharedIniFileCredentials()
+      creds.refresh (err) ->
+        expect(err.message).to.equal('Credentials not set in source_profile foo set in /home/user/.aws/credentials using profile default')
+      
+    it 'will fail if source profile config lacks credentials', ->
```

Comment:
Thanks for catching that! Duplication error on my part, I'll remove it.

Otherwise, unless you have any other tests you'd like added or changed, I plan on merging this today!